src/App.tsx

```diff
@@ -1,24 +1,57 @@
-import React, { ReactElement } from "react";
+import React, { ReactElement, useState } from "react";
 import Transformation from "./Transformation";
 import { SavedTransformation } from "./transformation-components/types";
 import { initPhone } from "./utils/codapPhone";
+import "./App.css";
 
-// This should be a pure function (ie: only render once)
 export const App = (): ReactElement => {
-  const parsedUrl = new URL(window.location.href);
+  const [loading, setLoading] = useState<boolean>(true);
+  const loadingMsg = <p className="loading">Connecting to CODAP...</p>;
+
+  const [initError, setInitError] = useState<boolean>(false);
```

Comment:
Instead of having two pieces of state that encode the loading/ready status, could we just have something like
```js
type ReadyStatus = "loading" | "ready" | "error";
const [status, setStatus] = useState<ReadyStatus>("loading");
```
That way we don't accidentally make changes in the future that result in and invalid set of state like `loading=false` and `initError = true`.

Reply:
Nice, this is much slicker. Fixed